include/deal.II/base/config.h.in

```diff
@@ -362,6 +362,21 @@
 #define DEAL_II_NAMESPACE_OPEN namespace dealii {
 #define DEAL_II_NAMESPACE_CLOSE }
 
+/***********************************************************************
+ * Store the build configuration in a namespace called config to reduce the
+ * usage of macros in the codebase.
+ */
+DEAL_II_NAMESPACE_OPEN
+namespace config
+{
+#ifdef DEAL_II_WITH_64BIT_INDICES
+  constexpr bool with_64bit_indices = true;
```

Comment:
This can be done more easily if you write this without the `#ifdef` as 
```
  constexpr book with_64bit_indices = @DEAL_II_WITH_64BIT_INDICES_BOOL@;
```
and in cmake define `DEAL_II_WITH_64BIT_INDICES_BOOL` in a way so that it gets replaces when doing a `CONFIGURE_FILE` (which this file undergoes).